zerver/models.py

```diff
@@ -93,7 +93,7 @@ def get_display_recipient_remote_cache(recipient_id, recipient_type, recipient_t
     # We don't really care what the ordering is, just that it's deterministic.
     user_profile_list = (UserProfile.objects.filter(subscription__recipient_id=recipient_id)
                                             .select_related()
-                                            .order_by('email'))
+                                            .order_by('id'))
```

Comment:
OK, makes sense!  Next time, break this out as a separate commit with that explanation, since it should be reviewed separately from the rest..